app/measure/measure.js

```diff
@@ -82,17 +82,17 @@ angular.module('Measure.Measure', ['ngRoute'])
 
       if (currentPhase == PHASE_UPLOAD) {
         $scope.currentPhase = gettextCatalog.getString('Upload');
-        $scope.currentSpeed = uploadSpeedVal ? getJustfiedSpeed(uploadSpeedVal) + ' ' + getSpeedUnit(uploadSpeedVal) : gettextCatalog.getString('Initializing');
+        $scope.currentSpeed = uploadSpeedVal ? getJustfiedSpeed(uploadSpeedVal) + ' Mb/s' : gettextCatalog.getString('Initializing');
       } else if (currentPhase == PHASE_DOWNLOAD) {
         $scope.currentPhase = gettextCatalog.getString('Download');
-        $scope.currentSpeed = downloadSpeedVal ? getJustfiedSpeed(downloadSpeedVal) + ' ' + getSpeedUnit(downloadSpeedVal) : gettextCatalog.getString('Initializing');
+        $scope.currentSpeed = downloadSpeedVal ? getJustfiedSpeed(downloadSpeedVal) + ' Mb/s' : gettextCatalog.getString('Initializing');
```

Comment:
getJustifiedSpeed() is the wrong thing to use if you want to fix units in Mb/s.  I suggest rewriting it to always return the speed in Mb/s plus at most two decimal points.

Reply:
@pboothe: This should be fixed now. PTAL?